layout-normalizer/src/gpos/marks.rs

```diff
@@ -35,6 +42,31 @@ impl PrintNames for MarkAttachmentRule {
     }
 }
 
+impl PrintNames for MarkLigaRule {
+    fn fmt_names(&self, f: &mut std::fmt::Formatter<'_>, names: &NameMap) -> std::fmt::Result {
+        let base_name = names.get(self.base);
+        write!(f, "{base_name} (lig) [")?;
+        for (i, anchor) in self.base_anchors.iter().enumerate() {
+            if i != 0 {
+                write!(f, ", ")?;
+            }
+            match anchor {
+                Some(a) => write!(f, "{a}"),
+                None => write!(f, "<NULL>"),
+            }?
+        }
+        writeln!(f, "]")?;
```

Comment:
this looks suspiciously likely to be a repetitive pattern, if it is perhaps we could share a fn?

Reply:
if we end up repeating it we can always refactor, but this is the last mark attachment rule so I don't anticipate that happening.